process/block/baseProcess.go

```diff
@@ -116,9 +116,11 @@ type baseProcessor struct {
 	processedMiniBlocksTracker    process.ProcessedMiniBlocksTracker
 	receiptsRepository            receiptsRepository
 
-	mutNonceOfFirstCommittedBlock sync.RWMutex
-	nonceOfFirstCommittedBlock    core.OptionalUint64
-	requestMissingHeadersFunc     func(missingNonces []uint64, shardID uint32)
+	mutNonceOfFirstCommittedBlock        sync.RWMutex
+	nonceOfFirstCommittedBlock           core.OptionalUint64
+	requestMissingHeadersFunc            func(missingNonces []uint64, shardID uint32)
+	cleanupBlockTrackerPoolsForShardFunc func(shardID uint32, noncesToPrevFinal uint64)
```

Comment:
these 2 functions could be incapsulated in an interface and passed as an object to the baseProcess.

Reply:
Yes, this place and many others could be refactor to work this way. Those new pointers that I've added will be removed anyway in future tasks, where the `shardCoordinator` will return correct shardID